### ISSUE TITLE
Roll the tape - recovering '90s data tapes in BitCurator

### ISSUE THREAD

**Issue by bitsgalore** [bitsgalore/bitsgalore.github.io#18]
<https://www.bitsgalore.org/2019/01/31/roll-the-tape-recovering-90s-data-tapes-in-bitcurator>

**Comment by paulyc** [comment]
Well if you haven't solved it yet check the tape density and block size settings for restore as they may be incorrect/not written to/unreadable from/written by a different version of dump/flipped by cosmic rays/whatever since it doesn't sound like it was ever too well-documented of a format! And when there's no standard it's virtually certain that every different flavor of Unix will make a different set of design decisions than every other flavor of Unix. (I admittedly had not heard of it until today so unsure of what kind of error syndrome that could be, whether user or developer or machine error..)

But likely not a coincidence they are all the same format, and unlikely that 7 tapes coincidentally of the same format would all be damaged to the point of being unreadable, but then that is possible too if they all came from the same bad batch of tapes, or were stored next to the same giant magnet, or maybe they were simply formatted and never written to in the first place! Who knows! With tape the possibilities are endless... Some hold secrets only an electron microscope can tell... (Actually I think they all do if they were ever written to more than once)

**Comment by bitsgalore** [comment]
@paulyc Thanks for the suggestions. I'm pretty sure block size isn't an issue here, since in that case tapeimgr would have reported some error in the block size estimation process. The formatted / never written to possibility sounds like a very plausible one though.

**Comment by Bstagnaro** [comment]
Hi Johan. Very interesting article. I came to it because I have a similar situation with the dds4 tape from 2002 and I don't know what program was used to make the backup. I would like to follow the steps you indicate, but I don't use linux. I wanted to ask you if there is any way to perform a similar procedure from a Windows or Mac environment. Thank you very much!

**Comment by bitsgalore** [comment]
@Bstagnaro Afraid not. The main reason I used a Linux-based workflow is that Linux natively supports SCSI devices, and already includes the low-level tools for reading data from tape (dd, mt). This makes reading tapes on a Linux system relatively straightforward. I don't know if modern Windows versions even support SCSI anymore, but even if they do you'd still need someting like dd and mt to work with the tape.

MacOs is based on Linux, but below thread suggests that it it doesn't include the required drivers (nor the mt tool):

<https://apple.stackexchange.com/questions/380390/is-it-possible-to-install-the-gnu-mt-tape-drive-command-in-osx>

The thread also links to a driver, but the author of that software explicitly warns that it's not production-ready. You could try installing Linux (e.g. Ubuntu or Linux Mint) on an old unused PC and try to work with that. Would probably be easier than trying to make it work with Windows or MacOS.

**Comment by Bstagnaro** [comment]
Thanks for the answer, I will try to install Lynux and follow these steps. Apparently this material was saved using the Retrospect application in a Mac environment, but it is not known for sure. Is there any information that can be retrieved from the start of the tape to confirm the program and version? Thank you very much again!

**Comment by bitsgalore** [comment]
@Bstagnaro No idea, as I'm not familiar with Retrospect myself. In any case, the tapeimgr-based workflow is completely format agnostic, so it should work with Retrospect tapes. However it'll then be up to you to work out how to further process the recovered files. A quick Google search turned up [this](https://www.retrospect.com/au/support/kb/tape_capacity_faq), which mentions:

> Retrospect compresses files into its own proprietary format using a "lossless" compression process. 

Because of this you might need the original Retrospect software for this (or any other software that might be able to deal with it).

**Comment by Bstagnaro** [comment]
ok, perfect! I'll try. Thanks again!

**Comment by Bstagnaro** [comment]
Hello! One more question (sorry but I am totally new to this) ... through Linux I can recover a file that was originally recorded in Mac os 9 classic, then modifying the extension, right?

**Comment by bitsgalore** [comment]
@Bstagnaro Yes, this shouldn't be any problem. BTW if you haven't seen this already [my 2019 iPres paper](https://www.bitsgalore.org/2019/09/09/recovering-90s-data-tapes-experiences-kb-web-archaeology) gives some additional details on the post-processing of recovered files (in your case you'll need to adapt the extraction step to the Retrospect format).

**Comment by Bstagnaro** [comment]
Perfect, I'm going to try! Thank you very much for all the information, then I'll tell you how it went! Thank you!
